tests/js/session-tests.js

```diff
@@ -646,4 +646,67 @@ module.exports = {
                 });
             });
     },
+
+    testPartialSync() {
+        if (!isNodeProccess) {
+            return Promise.resolve();
+        }
+
+        const username1 = uuid();
+        const username2 = uuid();
+        const realmName = uuid();
+
+        return runOutOfProcess(__dirname + '/download-api-helper.js', username1, realmName, REALM_MODULE_PATH)
+            .then(() => {
+                Realm.Sync.User.login('http://localhost:9080', username1, 'password').then(user1 => {
+                    TestCase.assertDefined(user1, 'user1');
+                    return new Promise((resolve, reject) => {
+                        let config1 = {
+                            sync: {
+                                user: user1,
+                                url: `realm://localhost:9080/~/${realmName}`,
+                            },
+                            schema: [{ name: 'Integer', properties: { value: 'int' } }],
+                        };
+
+                        return Realm.open(config1)
+                            .then(realm1 => {
+                                for(let i = 0; i < 10; i++) {
+                                    realm1.write(() => {
+                                        realm1.create('Integer', {value: i});
+                                    });
+                                }
+                            })
+                    })
+                    user1.logout();
+                })
+            })
+            .then(() => {
+                Realm.Sync.User.login('http://localhost:9080', username1, 'password').then(user2 => {
+                    TestCase.assertDefined(user2, 'user2');
+                    return new Promise((resolve, reject) => {
+                        let config2 = {
+                            sync: {
+                                user: user2,
+                                url: `realm://localhost:9080/~/${realmName}`,
+                                partial: true,
+                            },
+                            schema: [{ name: 'Integer', properties: { value: 'int' } }],
+                        };
+
+                        const realm2 = new Realm(config2);
+                        realm2.subscribeToObjects('Integer', 'value > 5').then((results, error) => {
```

Comment:
is this test really testing the partial sync. Seems to me it tests that realm.write did write more objects the the local Realm. I think real tests should be if an out of process client write some objects with type X and Y then the local already synced realm should receive only the X ones. So all the X should be present  and  the Y's should not be present

Reply:
The test logs in, creates 10 objects, and logs out. Then it logs in, set up partial sync and receive 4 objects. Or at least, that's what it is supposed to do :smile:

Reply:
Notice that the URI of the Realm is not the same when `partial` is `true` (it is rewritten in object store).

Reply:
Shouldn't you wait for upload to complete after creating the objects. Additionally, might be a good idea to delete the Realm after the user logs out and then wait for download to complete after opening the realm the second time.

Reply:
the url seems the same to me. Looks like someone had worked over the weekend. Let's find who :)

Reply:
At least in .NET though, the file path for partial and non-partial realms is the same, so they resolve to the same path on disk (which is why I suggested deleting the Realm).

Reply:
yes I like this too. Clear state = clear test results. It seems the test needs some touch up.

Reply:
I clearly misunderstood https://github.com/realm/realm-object-store/blob/master/src/sync/sync_config.hpp#L106